build-tools/scripts/y2metainfo

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env ruby
+#
+# y2metainfo - Automagically generate appstream data for YaST modules based on their desktop files
+#
+# Author: Stasiek Michalski <hellcp@opensuse.org>
+# Copyright 2019
+#
+
+require "rexml/document"
+require "optparse"
+require "inifile"
+
+options = {}
+
+OptionParser.new do |parser|
+  parser.banner = "Usage: y2metainfo [options]"
+  parser.on("-h", "--help", "Show this help message") do
+    puts parser
+  end
+  parser.on("-f", "--file FILEPATH", "The desktop file of the module") do |v|
+    options[:file] = v
+  end
+  parser.on("-l", "--license LICENSE", "The license of the module") do |v|
+    options[:license] = v
+  end
+  parser.on("-o", "--output DIRECTORY", "The output directory") do |v|
+    options[:output] = v
+  end
+end.parse!
+
+def metainfo_gen(desktop, license)
+  output = ""
+  xml = REXML::Document.new
+  file = IniFile.load(desktop, comment: nil)
+  data = file["Desktop Entry"]
+
+  xml << REXML::XMLDecl.new("1.0", "UTF-8")
+  base = xml.add_element("component", "type" => "addon", "xmlns" => "https://specifications.freedesktop.org/metainfo/1.0")
+  base.add_element("id").add_text(File.basename(desktop, ".desktop"))
+  base.add_element("launchable", "type" => "desktop-id").add_text(File.basename(desktop))
```

Comment:
is it correct? I expect parsing or desktop `exec` here. Naming desktop according to client is just convention, that is not true for some modules.

Reply:
launchable with type desktop-id takes desktop file and executes it, appstream launchable doesn't have type property `yast` (just like it has property `cockpit`), which is something I'm planning to propose for the next version of the spec.

Reply:
thanks for explanation. Could you please add it also to source code? So when we need to touch it, we know how it works.